### PR TITLE
Changing TUNING semantics to 'only change when specified'. 

### PR DESCRIPTION
**Pull request by vkip**
Achieving no impact on next time step needs to be handled separately.

### CODE REVIEW

**Review by bska — COMMENTED** [review]
I think this is good, but just for sake of discussion, I'd like to present an alternative implementation.  I'll use `XXXMBE` as my example.

You can use `if`-with-init to write these as
```C++
if (const auto& xxxmbe = record2.getItem("XXXMBE"); ! xxxmbe.defaultApplied(0)) {
    tuning.XXXMBE = xxxmbe.get<double>(0);
}
```
I have not made up my mind on what alternative I like better.  The `if`-with-init approach tends to take more lines of code and that's somewhat at a premium here.

**Comment by vkip** [comment]
> I have not made up my mind on what alternative I like better. The `if`-with-init approach tends to take more lines of code and that's somewhat at a premium here.

Seems I have a slight preference for fewer lines, but no strong opinion really. I'll  update if you decide that's better.

**Comment by bska** [comment]
> > I have not made up my mind on what alternative I like better. The `if`-with-init approach tends to take more lines of code and that's somewhat at a premium here.
> 
> Seems I have a slight preference for fewer lines

I think I agree with you here.  If there were only a couple of instances I might prefer `if`-with-init, but since we're dealing with more than 30 items the number of code statements does matter and explicit conditionals would probably obscure the intention.  I'll run a build check.

**Comment by bska** [comment]
jenkins build this please

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp [hunk 1]
```diff
@@ -929,71 +929,87 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
         const auto numrecords = handlerContext.keyword.size();
         auto tuning = this->snapshots.back().tuning();
 
+        // local helpers
+        auto nondefault_or_previous_double = [](const Opm::DeckRecord& rec, const std::string& item_name, double previous_value) {
```
[the remark this answers is not in the document]
Comment (bska) [reply src/opm/input/eclipse/Schedule/KeywordHandlers.cpp:933]: > `auto previous_value, .get<decltype(previous_value)>)` would save one explicit instance of the lambda

Sure, and we could have a general assignment helper function of the form
```C++
auto assign_if_specified = [](const DeckRecord& rec, const string& name, auto value, double& field)
{
    if (const auto& item = rec.getItem(name); !item.defaultApplied(0) {
        field = (item.*value)(0); // or field = std::invoke(value, item, 0);
    }
};
```
which we call as
```C++
assign_if_specified(rec, "TSINIT", &DeckItem::getSIDouble, tuning.TSINIT);
// ...
assign_if_specified(rec, "XXXMBE", &DeckItem::get<double>, tuning.XXXMBE);
```
but I'm not sure if that aids in clarity

**Review by bska — APPROVED** [review]
Thanks.  I'll merge into master.

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bska) [reply src/opm/input/eclipse/Schedule/KeywordHandlers.cpp:933]: > next time

I was being facetious.  I'm not actually suggesting we write code like this except for possibly a few situations that require a high degree of generic treatment.  It is of course standard C++, but pointers to member functions don't usually pop up outside of library code.

**Pull request merged**